### PR TITLE
ARM Diags V3

### PR DESCRIPTION
**Pull request by lannyzxj**
The ARM Data-oriented Metrics and Diagnostics Package for Climate Models (ARM-DIAGS) Verison 3:

• The CMIP6 model simulations are now the new default model reference in ARM-DIAGS-V3.
• ARM observation references extended to ENA and MAO sites with corresponding observations and model inputs.
• All ARM observation reprocessed & updated based on lastest value-added product data.
• The precipitation diurnal cycles are now added to the default output.
• New metrics of soil moisture at SGP and AOD at all sites.
• Season vertical CF and precipitation are extended to all four seasons.
• Input and output subfolders are now all site-dependent.
• Output HTML now fits the V3.
• Visualization modifications on the output figures.
• Minor fix on the python code terminal running indicators.
• Streamline the processes and modify all the source codes: detailed in src/XZ_src_development_log.txt

### CODE REVIEW

**Comment by chengzhuzhang** [comment]
@lannyzxj Thank you for the Pull Request! I will get to it in early next week.

**Review by EmmyChengTao — APPROVED** [review]

**Review by EmmyChengTao — APPROVED** [review]
I've tested this new branch ARM_Diags_V3 on my local machine and things look good on my end.

File: arm_diags/src/annual_cycle.py [hunk 1]
```diff
@@ -72,47 +93,23 @@ def annual_cycle_data(parameter):
...
-
-               #var24 = np.concatenate((var,var),axis=0)
+        obs_file = glob.glob(os.path.join(obs_path,sites[0][:3]+'armdiagsmon' + sites[0][3:5].upper()+'*c1.nc')) #read in monthly data
+    print('obs_file',obs_file)
+    fin = cdms2.open(obs_file[0])
+    for j, variable in enumerate(variables): 
+        try:
+            var = fin(variable)
+            #obs_var_season[j, :] = var_annual_cycle(var, seasons)
+            obs_var_season[j, :] = climo(var, 'ANNUALCYCLE')
```
Comment (chengzhuzhang) [inline arm_diags/src/annual_cycle.py:108]: @lannyzxj I noticed there is no longer necessary to convert the units of `pr`, `prw` and `tas` for this array. So i suppose the units conversion was taken care when producing the V3 datasets?

File: arm_diags/src/annual_cycle.py [hunk 2]
```diff
@@ -202,58 +207,67 @@ def annual_cycle_line_plot(parameter):
...
-        plt.legend(loc='best',prop={'size':15})
+        plt.title('Annual Cycle: Model vs OBS vs CMIP',fontsize=15)
+        plt.xlabel('Month',fontsize=15)
+        plt.legend(loc='best',prop={'size':12})
         plt.ylabel(var_longname[j])
-        fig.savefig(output_path+'/figures/'+variable+'_annual_cycle_'+sites[0]+'.png')
+        #special notes for models mistreating surface type [XZ]:
+        if (variable == 'hfls') or (variable == 'hfss') or (variable == 'rsus'):
+            if (sites[0] == 'enac1') or (sites[0] == 'twpc1') or (sites[0] == 'twpc2') or (sites[0] == 'twpc3'):
+                ax.text(0.5, 0.05,'Most GCMs treat the surface type as ocean', ha='center', va='center', transform=ax.transAxes,fontsize=10)
```
Comment (chengzhuzhang) [inline arm_diags/src/annual_cycle.py]: Suggest to change to `Note: the selected grid points were ocean grids in most GCMs.`

File: arm_diags/src/diurnal_cycle.py [hunk 3]
```diff
@@ -102,87 +142,131 @@ def diurnal_cycle_data(parameter):
...
-    
+
+            # plotting
             fig = plt.figure()# Create figure
-            ax  =fig.add_axes([0.15, 0.14, 0.8, 0.8]) # Create axes
-            xax_3hr=np.array([3.0*x-6.5 for x in range(8)])
-            xax_1hr=np.array([1.0*x-6.5 for x in range(24)]) 
+            ax  =fig.add_axes([0.10, 0.10, 0.85, 0.85]) # Create axes
+            #-----------------------------------------------------------------------------
+            # create site-dependent time offset for the local time & plotting range [XZ]
```
Comment (chengzhuzhang) [inline arm_diags/src/diurnal_cycle.py:257]: Thank you for fixing the site depended local time offset!

**Review by chengzhuzhang — COMMENTED** [review]
Hey @lannyzxj Thank you so much for bringing in these nice changes! I only have minor comments. Another two items we can work on as additional Pull Requests are:
Add pre-processing scripts in this prepository so that we can track the provenance of the data creation from original ARM data streams; and to use a more succinct fft function for compute first harmonics (example: https://github.com/E3SM-Project/e3sm_diags/blob/master/e3sm_diags/driver/utils/diurnal_cycle.py). I wish i had the time to implement both. Hope you could be able to work on this. Thank you!

**Comment by chengzhuzhang** [comment]
Since convection onset statistics were not documented in the literature for NSA, I think we can leave it out for now.

File: arm_diags/src/annual_cycle.py (hunk 1 above)
Comment (lannyzxj) [reply arm_diags/src/annual_cycle.py:108]: The unit conversions are performed in function 'climo' inside the 'utils.py' as originally in V2, and I keep it that way. The original units of pr, prw, and tas are the same as V2.

File: arm_diags/src/annual_cycle.py (hunk 2 above)
Comment (lannyzxj) [reply arm_diags/src/annual_cycle.py]: Thanks, the note is changed.

File: arm_diags/src/diurnal_cycle.py (hunk 3 above)
Comment (lannyzxj) [reply arm_diags/src/diurnal_cycle.py:257]: You're welcome!

**Comment by lannyzxj** [comment]
>

**Comment by lannyzxj** [comment]
> Since convection onset statistics were not documented in the literature for NSA, I think we can leave it out for now.

The NSA is now removed in V3.

**Comment by chengzhuzhang** [comment]
Hey Xiaojian, you could go ahead and Merge your Pull Request through "Squash and Merge". For the other two possible enhancement mentioned here: https://github.com/ARM-DOE/arm-gcm-diagnostics/pull/26#pullrequestreview-860065144 You could use new new Pull Requests. Thanks.

**Comment by lannyzxj** [comment]
> Hey Xiaojian, you could go ahead and Merge your Pull Request through "Squash and Merge". For the other two possible enhancement mentioned here: [#26 (review)](https://github.com/ARM-DOE/arm-gcm-diagnostics/pull/26#pullrequestreview-860065144) You could use new new Pull Requests. Thanks.

Dear Dr. Zhang, thanks! I will start merging, and further work on the other enhancements.

**Pull request merged**